cpp/src/arrow/util/basic_decimal.h

```diff
@@ -156,8 +162,13 @@ class ARROW_EXPORT BasicDecimal128 {
   static const BasicDecimal128& GetMaxValue();
 
  private:
+#if ARROW_LITTLE_ENDIAN
```

Comment:
I think we can use `std::array<uint64_t, 2>` here. Maybe unify `Decimal128` and `Decimal256`.